comp/api/api/apiimpl/internal/agent/agent_test.go

```diff
@@ -63,3 +70,447 @@ func TestSetupHandlers(t *testing.T) {
 		assert.Equal(t, tc.wantCode, resp.StatusCode, "%s %s failed with a %d, want %d", tc.method, tc.route, resp.StatusCode, tc.wantCode)
 	}
 }
+
+func TestInstallInfoAPIRoutes(t *testing.T) {
+	// Backup original environment
+	originalTool := os.Getenv("DD_INSTALL_INFO_TOOL")
+	originalToolVersion := os.Getenv("DD_INSTALL_INFO_TOOL_VERSION")
+	originalInstallerVersion := os.Getenv("DD_INSTALL_INFO_INSTALLER_VERSION")
+
+	os.Setenv("DD_INSTALL_INFO_TOOL", "test-tool")
+	os.Setenv("DD_INSTALL_INFO_TOOL_VERSION", "1.0.0")
+	os.Setenv("DD_INSTALL_INFO_INSTALLER_VERSION", "test-installer-1.0")
+
+	defer func() {
+		if originalTool == "" {
+			os.Unsetenv("DD_INSTALL_INFO_TOOL")
+		} else {
+			os.Setenv("DD_INSTALL_INFO_TOOL", originalTool)
+		}
+		if originalToolVersion == "" {
+			os.Unsetenv("DD_INSTALL_INFO_TOOL_VERSION")
+		} else {
+			os.Setenv("DD_INSTALL_INFO_TOOL_VERSION", originalToolVersion)
+		}
+		if originalInstallerVersion == "" {
+			os.Unsetenv("DD_INSTALL_INFO_INSTALLER_VERSION")
+		} else {
+			os.Setenv("DD_INSTALL_INFO_INSTALLER_VERSION", originalInstallerVersion)
+		}
+	}()
+
+	router := setupRoutes()
+	ts := httptest.NewServer(router)
+	defer ts.Close()
+
+	tests := []struct {
+		name         string
+		route        string
+		method       string
+		payload      interface{}
+		expectedCode int
+		description  string
+		assertFunc   func(t *testing.T, resp *http.Response, body []byte)
+	}{
+		{
+			name:         "install info get with GET method",
+			route:        "/install-info",
+			method:       "GET",
+			payload:      nil,
+			expectedCode: 200,
+			description:  "GET request to install-info should succeed with test config",
+			assertFunc: func(t *testing.T, resp *http.Response, body []byte) {
+				assert.Equal(t, "application/json", resp.Header.Get("Content-Type"))
+				var installInfo installinfo.InstallInfo
+				err := json.Unmarshal(body, &installInfo)
+				assert.NoError(t, err)
+				assert.Equal(t, "test-tool", installInfo.Tool)
+				assert.Equal(t, "1.0.0", installInfo.ToolVersion)
+				assert.Equal(t, "test-installer-1.0", installInfo.InstallerVersion)
+			},
+		},
+		{
+			name:         "install info get with POST method",
+			route:        "/install-info",
+			method:       "POST",
+			payload:      nil,
+			expectedCode: 400,
+			description:  "POST request to install-info should be rejected",
```

Comment:
The description for this test and the one below might miss some information on why they are rejected. Are they rejected because the body is empty? If so, it would be good to add that as part of the description. 😄